infra/omop_cdm.bicep

```diff
@@ -0,0 +1,192 @@
+param branchName string
+
+@description('The name of the postgres server')
+param postgresServerName string
+
+@secure()
+@description('The URL of the container where the CDM data is stored')
+param cdmContainerUrl string
+
+@secure()
+@description('The SAS token to access the CDM data')
+param cdmSasToken string
+
+@secure()
+@description('The name of the OMOP CDM database')
+param postgresOMOPCDMDatabaseName string
+
+@secure()
+@description('Admin password for the postgres server')
+param postgresAdminPassword string
+
+@secure()
+@description('Admin password for the webapi database')
+param postgresWebapiAdminPassword string
+
+@secure()
+@description('Password for the cdm user')
+param postgresOMOPCDMpassword string
+
+@secure()
+@description('The name of the keyvault')
+param keyVaultName string
+
+var postgresOMOPCDMSchemaName = 'cdm'
+var postgresOMOPVocabularySchemaName = 'vocabulary'
+var postgresOMOPResultsSchemaName = 'cdm_results'
+var postgresOMOPTempSchemaName = 'temp'
+var postgresOMOPCDMRole = 'cdm_reader'
+var postgresOMOPCDMUsername = 'cdm_user'
+var postgresOMOPCDMUserSecretName = '${postgresOMOPCDMDatabaseName}-cdm-user-password'
+var postgresOMOPCDMJDBCConnectionStringSecretName = '${postgresOMOPCDMDatabaseName}-cdm-jdbc-connection-string'
+var postgresWebAPISchemaName = 'webapi'
+var postgresWebapiAdminUsername = 'ohdsi_admin_user'
+var postgresAdminUsername = 'postgres_admin'
+var postgresOMOPCDMJDBCConnectionString = 'jdbc:postgresql://${postgresServer.properties.fullyQualifiedDomainName}:5432/${postgresOMOPCDMDatabaseName}?user=${postgresOMOPCDMUsername}&password=${postgresOMOPCDMpassword}&sslmode=require'
+
+// Get the postgres server
+resource postgresServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-12-01' existing = {
+  name: postgresServerName
+}
+
+// Get the keyvault
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' existing = {
+  name: keyVaultName
+}
+
+// Store OMOM CDM user password in keyvault
+resource postgresAdminSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: postgresOMOPCDMUserSecretName
+  parent: keyVault
+  properties: {
+    value: postgresOMOPCDMpassword
+  }
+}
+
+// Store the OMOP CDM JDBC connection string in keyvault
+resource postgresOMOPCDMJDBCConnectionStringSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: postgresOMOPCDMJDBCConnectionStringSecretName
+  parent: keyVault
+  properties: {
+    value: postgresOMOPCDMJDBCConnectionString
+  }
+}
+
+// Create a new PostgreSQL database for the OMOP CDM
+resource postgresDatabase 'Microsoft.DBforPostgreSQL/flexibleServers/databases@2022-12-01' = {
+  name: postgresOMOPCDMDatabaseName
+  parent: postgresServer
+  properties: {
+    charset: 'utf8'
+    collation: 'en_US.utf8'
+  }
+}
+
+// Create a managed identity for the deployment script
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: 'managed-identity-omop-cdm'
```

Comment:
Maybe we can use the same identity that was already created for running the previous sql scripts?

Reply:
I think we should refactor our modules structure in the next iteration,
maybe this shared part should be in main along with some other shared components,
unless this identity should be deprovisioned with the script when it finished.